bionic/dagviz.py

```diff
@@ -91,14 +91,39 @@ def dot_from_graph(graph, vertical=False, curvy_lines=False, name=None):
         rankdir="TB" if vertical else "LR",
     )
 
-    node_lists_by_cluster = defaultdict(list)
+    # First, we cluster together any nodes that share an entity name. This includes
+    # nodes generated by a common tuple function, and nodes that differ only by case
+    # key.
+    node_clusters_by_entity_name = defaultdict(list)
     for node in graph.nodes():
-        descriptor = graph.nodes[node]["descriptor"]
-        node_lists_by_cluster[descriptor].append(node)
-
-    descriptors = list(set(graph.nodes[node]["descriptor"] for node in graph.nodes()))
-    color_strs_by_descriptor = hpluv_color_dict(
-        descriptors, saturation=99, lightness=90
+        entity_names = list(node.dnode.all_entity_names())
+        # If this node has no entity names at all, we'll pretend it has a single entity
+        # name. (This is unlikely to happen in practice, but could happen for a
+        # descriptor like the empty tuple `(,)`.)
+        if len(entity_names) == "":
```

Comment:
I think this should be `len(entity_names) == 0`

Reply:
Yup! Good catch.